python/pyarrow/tests/test_pandas.py

```diff
@@ -4542,3 +4569,48 @@ def test_does_not_mutate_timedelta_nested():
     df = table.to_pandas()
 
     assert df["timedelta_2"][0].to_pytimedelta() == timedelta_2[0]
+
+
+def test_roundtrip_nested_map_table_with_pydicts():
```

Comment:
This is a bit paranoid, but could you also test the `to_pandas` conversion on a MapArray that has been sliced (or even a chunked one where each chunk has been sliced)? That often brings out some bugs in these implementations.

Reply:
Added some additional tests in 0939fa65ab99e709f0ec4f3bf2609f4f353b218b

The actual scenario was a bit difficult to generate. Let me know if the tests are sufficient or if they need more refinement.